### PR TITLE
Fix optional_kernel_features tests

### PR DESCRIPTION
**Pull request by aleksmesh**
Added kernel functions with accessor to avoid compiler optimisation. Removed single_task call and change parallel_for to use nd_range instead of range for kernels with specicfied group size attributes.

### CODE REVIEW

File: tests/optional_kernel_features/kernel_features_common.h [hunk 1]
```diff
@@ -126,21 +164,46 @@ class non_decorated_call_non_decorated_function {
   }
 };
 
+/**
+ * @brief Not decorated functor that invokes not decorated function that use
+ * feature defined in FeatureTypeT
```
Comment (keryell) [inline tests/optional_kernel_features/kernel_features_common.h:169]:
```suggestion
   @brief Non-decorated functor that invokes non-decorated function that uses
   feature defined in FeatureTypeT
```

File: tests/optional_kernel_features/kernel_features_common.h [hunk 2]
```diff
@@ -38,7 +49,7 @@ namespace kernel_features_common {
  */
 template <typename T, sycl::aspect aspect>
 [[sycl::device_has(aspect)]] SYCL_EXTERNAL void
-use_feature_function_external_decorated();
+use_feature_function_external_decorated(const sycl::accessor<bool, 1> &acc);
```
Comment (keryell) [inline tests/optional_kernel_features/kernel_features_common.h:52]: I thought that the coding style for SYCL and the CTS is to have `&` and `*` part of the type everywhere.
```suggestion
use_feature_function_external_decorated(const sycl::accessor<bool, 1>& acc);
```

File: tests/optional_kernel_features/kernel_features_common.h [hunk 3]
```diff
@@ -255,6 +332,118 @@ constexpr sycl::aspect get_another_aspect() {
...
+      use_feature_function_decorated<FeatureTypeT, FeatureAspectT>(acc);
+    };
+  } else if constexpr (CallType == call_attribute_type::dummy_decorated) {
+    return [=] { dummy_function_decorated<FeatureAspectT>(acc); };
+  } else if constexpr (CallType == call_attribute_type::dummy_non_decorated) {
+    return [=]() { dummy_function_non_decorated(acc); };
+  } else if constexpr (CallType == call_attribute_type::type_used) {
+    return [acc]() { USE_FEATURE(FeatureTypeT); };
+  } else if constexpr (CallType == call_attribute_type::type_used_with_attr) {
+    return [=]() [[sycl::device_has(AnotherFeatureAspect)]] {
```
Comment (keryell) [inline tests/optional_kernel_features/kernel_features_common.h:366]: Useless
```suggestion
    return [=] {
      use_feature_function_external_decorated<FeatureTypeT, FeatureAspectT>(
          acc);
    };
  } else if constexpr (CallType ==
                       call_attribute_type::external_decorated_with_attr) {
    return [acc]() [[sycl::device_has(AnotherFeatureAspect)]] {
      use_feature_function_external_decorated<FeatureTypeT, FeatureAspectT>(
          acc);
    };
  } else if constexpr (CallType == call_attribute_type::non_decorated) {
    return [=] {
      use_feature_function_non_decorated_with_accessor<FeatureTypeT>(acc);
    };
  } else if constexpr (CallType == call_attribute_type::decorated) {
    return [=] {
      use_feature_function_decorated<FeatureTypeT, FeatureAspectT>(acc);
    };
  } else if constexpr (CallType == call_attribute_type::dummy_decorated) {
    return [=] { dummy_function_decorated<FeatureAspectT>(acc); };
  } else if constexpr (CallType == call_attribute_type::dummy_non_decorated) {
    return [=] { dummy_function_non_decorated(acc); };
  } else if constexpr (CallType == call_attribute_type::type_used) {
    return [acc] { USE_FEATURE(FeatureTypeT); };
  } else if constexpr (CallType == call_attribute_type::type_used_with_attr) {
    return [=]() [[sycl::device_has(AnotherFeatureAspect)]] {
```

File: tests/optional_kernel_features/kernel_features_common.h [hunk 4]
```diff
@@ -494,17 +905,23 @@ class kernel_submission_call;
                                                                             \
   {                                                                         \
     auto single_task_action = [&QUEUE] {                                    \
+      bool value = false;                                                   \
```
Comment (keryell) [inline tests/optional_kernel_features/kernel_features_common.h:908]: Some day we will have to remove all these macros... :-(

**Review by keryell — COMMENTED** [review]
Thanks.

File: tests/optional_kernel_features/kernel_features_common.h (hunk 1 above)
Comment (aleksmesh) [reply tests/optional_kernel_features/kernel_features_common.h]: Fixed. Thanks!

File: tests/optional_kernel_features/kernel_features_common.h (hunk 2 above)
Comment (aleksmesh) [reply tests/optional_kernel_features/kernel_features_common.h:52]: Agree with you but clang-format.

File: tests/optional_kernel_features/kernel_features_common.h (hunk 3 above)
Comment (aleksmesh) [reply tests/optional_kernel_features/kernel_features_common.h]: Done. Thanks for suggestion.

File: tests/optional_kernel_features/kernel_features_common.h (hunk 4 above)
Comment (aleksmesh) [reply tests/optional_kernel_features/kernel_features_common.h:908]: I hope.

**Review by ProGTX — APPROVED** [review]

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**